radialgraph/src/main/java/com/duartbreedt/radialgraph/drawable/GraphDrawable.kt

```diff
@@ -29,11 +34,67 @@ abstract class GraphDrawable(
 
         return Paint().apply {
             strokeWidth = graphConfig.strokeWidth
-            color = state.color
             pathEffect = DashPathEffect(floatArrayOf(state.length!!, state.length!!), phase)
             style = Paint.Style.STROKE
             flags = Paint.ANTI_ALIAS_FLAG
             strokeCap = graphConfig.capStyle.paintCapStyle
+
+            if (!graphConfig.isGradientEnabled()) {
+                color = state.color
+            }
+
+            if (graphConfig.gradientType == GradientType.SWEEP) {
+                applySweepGradient(state, this, graphConfig.gradientFill)
+            }
+        }
+    }
+
+    private fun applySweepGradient(state: SectionState, paint: Paint, gradientFill: GradientFill) {
+        if (state.gradientColors.isNullOrEmpty()) {
+            paint.color = state.color
+            return
+        }
+
+        val colorList = mutableListOf(
+            state.color
+        )
+        colorList.addAll(state.gradientColors)
+
+        var positionList: MutableList<Float>? = null
+        if (gradientFill == GradientFill.SECTION) {
+            positionList = MutableList(colorList.size) { it.toFloat() }
+            //Get spacing between each gradient section
+            val gradientGap = (1f / (positionList.size -1)) * state.sweepSize
+
+            positionList = positionList.map {
+                (gradientGap * it) + state.startPosition
+            }.toMutableList()
+
+            //Fix for gradient overflow when using a cap style other than 'BUTT'
+            if (state.startPosition == 0f && graphConfig.capStyle != Cap.BUTT) {
+                colorList.add(state.color)
+                positionList.addAll(
+                    listOf(
+                        positionList.removeLast().coerceAtMost(endGradientCapFill),
+                        endGradientCapFill
+                    )
+                )
+            }
+        }
+
+        paint.shader = SweepGradient(
+            this@GraphDrawable.bounds.centerX().toFloat(),
+            this@GraphDrawable.bounds.centerY().toFloat(),
+            colorList.toIntArray(),
+            positionList?.toFloatArray()
```

Comment:
So, if we set it to null, it will space them out evenly as if it was painting the full 360 arc. (If you had an arc that was only 180 or 50 percent then the second color will on be shown partially, this may not be the desired before)

Added the `gradientFill` property to switch between those fill modes.